### ISSUE TITLE
Using ComponentType within libdef

### ISSUE THREAD

**Issue by dquang-tran** [facebook/flow#5649]
Getting  
```
48:     declare class A extends ComponentType<Props> {
                                      ^^^^^^^^^^^^^ ComponentType. type referenced from value position
  6:     import type { Node, ComponentType } from 'react';
                             ^^^^^^^^^^^^^ type ComponentType
```

My libdef code

```
import type { Node, ComponentType } from 'react';

declare module "libName" {

    declare type Props = {...}

    declare class A extends ComponentType<Props> {
       ...
    }
    declare export default typeof A;
    declare export var A: typeof A;
}

```

Can someone explain to me why this is invalid?

Having 
`declare class A extends React$Component<Props>` does not cause any issues, but I am curious why I can't use ComponentType

**Comment by idiostruct** [comment]
The extends keyword is used in class declarations or class expressions to create a class as a child of another class. `ComponentType` would need to be a class (i.e., `declare class React$ComponentType`) for it to be extensible. Does that make sense?

**Comment by gnprice** [comment]
This issue can be closed, because it was fixed in v0.143.0.

> My libdef code
> 
> ```
> import type { Node, ComponentType } from 'react';
>
> declare module "libName" {
> // …
> ```

This is invalid; rather, you have to put the import inside the `declare module` block.

Before v0.143.0, Flow didn't give an error for this, and instead had very confusing behavior like what's in this issue report. Now it gives a good error message. See #6414 for details.
